kani-compiler/src/kani_compiler.rs

```diff
@@ -121,18 +129,17 @@ enum CompilationStage {
         target_harnesses: Vec<HarnessId>,
         next_harnesses: Vec<Vec<HarnessId>>,
         all_harnesses: HashMap<HarnessId, HarnessInfo>,
+        crate_info: CrateInfo,
+    },
+    Done {
+        metadata: Option<(KaniMetadata, CrateInfo)>,
```

Comment:
Why the encoding with an optional metadata? From the other comments it seems this only occurs in like `--version`, e.g. if no actual compilation happens, so why not make it an explicit `CompilationStage::CompilationSkipped` variant?

Reply:
Yeah, I thought about that, but we still two other modes where we do run codegen and we don't have a metadata (tests and pubfn). So this would still need to be optional.

Once we migrate them, I agree that we should just have a different state.

Reply:
Actually, I think we can add this now. Let me do that. Thanks